src/app/services/toast.service.ts

```diff
@@ -28,37 +28,47 @@ export class ToastService {
   };
 
   toastEvents$: Observable<ToastEvent>;
-
-  toastLogs: ToastEventLog[] = [];
+  toastLogs$: Observable<ToastEventLog[]>;
 
   private _toastEvents = new Subject<ToastEvent>();
+  private _toastLogs$ = new BehaviorSubject<ToastEventLog[]>([]);
 
   constructor() {
     this.toastEvents$ = this._toastEvents.asObservable();
+    this.toastLogs$ = this._toastLogs$.asObservable();
   }
 
   showSuccess(header: string, body: string) {
     this._show(
       { header, body, type: Event.SUCCESS },
       { ...this.DEFAULT_CONFIG_SUCCESS }
     );
-    this.toastLogs.push({ type: Event.SUCCESS, time: this.getTime() });
+    // this.toastLogs.push({ type: Event.SUCCESS, time: this.getTime() });
+
+    const log: ToastEventLog = { type: Event.SUCCESS, time: this.getTime() };
+    this._toastLogs$.next([...this._toastLogs$.getValue().concat([log])]);
   }
 
```

Comment:
wanted to understand how I can make .next([...this._toastLogs$.getValue().concat([log])]) logic better (if my approach is right in the first place)

Reply:
Better how? In terms of readability?

If yes, then you can just use a spread operator `[...this._toastLogs$.getValue(), log]`